lib/collection/src/operations/payload_ops.rs

```diff
@@ -3,15 +3,16 @@ use segment::types::{Filter, Payload, PayloadKeyType, PointIdType};
 use serde;
 use serde::{Deserialize, Serialize};
 
+use super::point_ops::PointsSelector;
 use super::{split_iter_by_shard, OperationToShard, SplitByShard};
 use crate::hash_ring::HashRing;
 use crate::shards::shard::ShardId;
 
 #[derive(Debug, Deserialize, Serialize, JsonSchema, Clone)]
 pub struct SetPayload {
     pub payload: Payload,
-    /// Assigns payload to each point in this list
-    pub points: Vec<PointIdType>, // ToDo: replace with point selector
+    /// Assigns payload to each point selected
+    pub selected_points: PointsSelector,
```

Comment:
Should also be backward compatible. Either make `points` optional and add another optional `filter` field, or add selector enum field with an alias to `points`. Selector should be deserializable from `Vec<PointIdType>`  (I would prefer the first option here)